### PR TITLE
Adds endpoints for customer authentication via the JS SDK

### PR DESCRIPTION
**Pull request by andrew-waters**
Exposes additional endpoints via JS to allow users to authenticate and update against their accounts using tokens.

### CODE REVIEW

File: src/features/customer.coffee [hunk 1]
```diff
@@ -4,5 +4,28 @@
...
+
+  `// @endif
+  `
+
+  `// @if TARGET=='js'
+  `
+  class Customer
+
+    endpoint: 'customers'
+    constructor: (@m) ->
```
Comment (outrunthewolf) [inline src/features/customer.coffee]: Coding style: line between endpoint and constructor

File: src/features/customer.coffee [hunk 2]
```diff
@@ -4,5 +4,28 @@
...
+
+    Authenticate: (data, callback, error) ->
```
Comment (outrunthewolf) [inline src/features/customer.coffee]: Can we share functions between nodejs and js?

File: src/features/customer.coffee [hunk 3]
```diff
@@ -4,5 +4,28 @@
...
+
+      return @m.Request @endpoint+'/authenticate', 'POST', data, callback, error
+
+    Update: (token, data, callback, error) ->
```
Comment (outrunthewolf) [inline src/features/customer.coffee]: Are we replicating functionality here from abstract?

Comment (andrew-waters) [inline src/features/customer.coffee]: Yes, but only partially - we don't want all abstract methods implicitly exposed

Comment (outrunthewolf) [inline src/features/customer.coffee]: Talk to @jHoldroyd He understands all this better than my inferior mind

**Comment by machard** [comment]
Hi,

Is there any doc about the authenticate endpoint ?
By default there is no password field in the Customer Flow.

Thanks

**Comment by outrunthewolf** [comment]
@machard Moved to an issue rather than comment on the PR https://github.com/moltin/js-sdk/issues/24

**Pull request closed without merging**